File: CBTReframe/Models/AIProvider.swift
import Foundation

enum AIProvider: String, CaseIterable, Codable, Identifiable {
    case openai = "OpenAI"
    case anthropic = "Anthropic"
    case deepseek = "DeepSeek"
    case gemini = "Google Gemini"
    case kimi = "Kimi (Moonshot)"
    case local = "本地（离线）"

    var id: String { rawValue }

    var displayName: String { rawValue }

    var requiresAPIKey: Bool {
        self != .local
    }

    /// 网络拉取失败或未配置 Key 时的兜底列表。
    var fallbackModels: [AIModel] {
        switch self {
        case .openai:
            return [
                AIModel(id: "gpt-4.1", name: "GPT-4.1"),
                AIModel(id: "gpt-4.1-mini", name: "GPT-4.1 Mini"),
                AIModel(id: "gpt-4.1-nano", name: "GPT-4.1 Nano"),
                AIModel(id: "gpt-4o", name: "GPT-4o"),
                AIModel(id: "gpt-4o-mini", name: "GPT-4o Mini"),
            ]
        case .anthropic:
            return [
                AIModel(id: "claude-sonnet-4-20250514", name: "Claude Sonnet 4"),
                AIModel(id: "claude-3-5-haiku-20241022", name: "Claude 3.5 Haiku"),
            ]
        case .deepseek:
            return [
                AIModel(id: "deepseek-chat", name: "DeepSeek Chat"),
                AIModel(id: "deepseek-reasoner", name: "DeepSeek Reasoner"),
            ]
        case .gemini:
            return [
                AIModel(id: "gemini-2.0-flash", name: "Gemini 2.0 Flash"),
                AIModel(id: "gemini-2.0-flash-lite", name: "Gemini 2.0 Flash-Lite"),
                AIModel(id: "gemini-1.5-flash", name: "Gemini 1.5 Flash"),
                AIModel(id: "gemini-1.5-pro", name: "Gemini 1.5 Pro"),
            ]
        case .kimi:
            return [
                AIModel(id: "moonshot-v1-8k", name: "Moonshot v1 8K"),
                AIModel(id: "moonshot-v1-32k", name: "Moonshot v1 32K"),
                AIModel(id: "kimi-k2-turbo-preview", name: "Kimi K2 Turbo"),
                AIModel(id: "kimi-k2-thinking-preview", name: "Kimi K2 Thinking"),
            ]
        case .local:
            return [
                AIModel(id: "local", name: "内置分析"),
            ]
        }
    }

    var defaultModel: AIModel {
        fallbackModels[0]
    }

    var baseURL: String {
        switch self {
        case .openai:
            return "https://api.openai.com/v1/chat/completions"
        case .anthropic:
            return "https://api.anthropic.com/v1/messages"
        case .deepseek:
            return "https://api.deepseek.com/v1/chat/completions"
        case .gemini:
            return "https://generativelanguage.googleapis.com/v1beta"
        case .kimi:
            return "https://api.moonshot.cn/v1/chat/completions"
        case .local:
            return ""
        }
    }
}

struct AIModel: Codable, Identifiable, Hashable {
    let id: String
    let name: String
}

File: shared/src/commonMain/kotlin/com/henryliu/cbtreframe/shared/AIProvider.kt
package com.henryliu.cbtreframe.shared

import kotlinx.serialization.Serializable

enum class AIProvider {
    DEEPSEEK,
    OPENAI,
    ANTHROPIC,
    GEMINI,
    KIMI,
    LOCAL
}

@Serializable
data class AIModel(
    val provider: AIProvider,
    val modelName: String,
    val displayName: String,
    val isReasoning: Boolean = false,
    val isPremium: Boolean = false,
)

object ModelDisplayDictionary {
    private val DEEPSEEK_CHAT = AIModel(AIProvider.DEEPSEEK, "deepseek-chat", "DeepSeek Chat")
    private val DEEPSEEK_REASONER = AIModel(AIProvider.DEEPSEEK, "deepseek-reasoner", "DeepSeek Reasoner", isReasoning = true)
    private val GPT_4O = AIModel(AIProvider.OPENAI, "gpt-4o", "GPT-4o")
    private val GPT_4O_MINI = AIModel(AIProvider.OPENAI, "gpt-4o-mini", "GPT-4o Mini")
    private val GPT_4_1 = AIModel(AIProvider.OPENAI, "gpt-4.1", "GPT-4.1")
    private val GPT_4_1_MINI = AIModel(AIProvider.OPENAI, "gpt-4.1-mini", "GPT-4.1 Mini")
    private val GPT_4_1_NANO = AIModel(AIProvider.OPENAI, "gpt-4.1-nano", "GPT-4.1 Nano")
    private val CLAUDE_SONNET_4 = AIModel(AIProvider.ANTHROPIC, "claude-sonnet-4-20250514", "Claude Sonnet 4")
    private val CLAUDE_3_5_HAIKU = AIModel(AIProvider.ANTHROPIC, "claude-3-5-haiku-20241022", "Claude 3.5 Haiku")
    private val GEMINI_FLASH_LATEST = AIModel(AIProvider.GEMINI, "gemini-flash-latest", "Gemini Flash Latest")
    private val GEMINI_2_5_FLASH = AIModel(AIProvider.GEMINI, "gemini-2.5-flash", "Gemini 2.5 Flash")
    private val GEMINI_2_0_FLASH = AIModel(AIProvider.GEMINI, "gemini-2.0-flash", "Gemini 2.0 Flash")
    private val GEMINI_2_0_FLASH_LITE = AIModel(AIProvider.GEMINI, "gemini-2.0-flash-lite", "Gemini 2.0 Flash-Lite")
    private val GEMINI_1_5_PRO = AIModel(AIProvider.GEMINI, "gemini-1.5-pro", "Gemini 1.5 Pro", isPremium = true)
    private val GEMINI_1_5_FLASH = AIModel(AIProvider.GEMINI, "gemini-1.5-flash", "Gemini 1.5 Flash")
    private val MOONSHOT_V1_8K = AIModel(AIProvider.KIMI, "moonshot-v1-8k", "Moonshot v1 8K")
    private val MOONSHOT_V1_32K = AIModel(AIProvider.KIMI, "moonshot-v1-32k", "Moonshot v1 32K")
    private val KIMI_K2_TURBO = AIModel(AIProvider.KIMI, "kimi-k2-turbo-preview", "Kimi K2 Turbo")
    private val KIMI_K2_THINKING = AIModel(AIProvider.KIMI, "kimi-k2-thinking-preview", "Kimi K2 Thinking", isReasoning = true)
    val LOCAL_BUILTIN = AIModel(AIProvider.LOCAL, "local", "内置分析")

    val entries = listOf(
        DEEPSEEK_CHAT,
        DEEPSEEK_REASONER,
        GPT_4O,
        GPT_4O_MINI,
        GPT_4_1,
        GPT_4_1_MINI,
        GPT_4_1_NANO,
        CLAUDE_SONNET_4,
        CLAUDE_3_5_HAIKU,
        GEMINI_FLASH_LATEST,
        GEMINI_2_5_FLASH,
        GEMINI_2_0_FLASH,
        GEMINI_2_0_FLASH_LITE,
        GEMINI_1_5_PRO,
        GEMINI_1_5_FLASH,
        MOONSHOT_V1_8K,
        MOONSHOT_V1_32K,
        KIMI_K2_TURBO,
        KIMI_K2_THINKING,
        LOCAL_BUILTIN
    )

    fun getDisplayName(provider: AIProvider, modelId: String): String? {
        return entries.firstOrNull { it.provider == provider && it.modelName.equals(modelId, ignoreCase = true) }?.displayName
    }
}

fun prettyGenericName(id: String): String {
    return id.split("-").joinToString(" ") { part ->
        part.replaceFirstChar { if (it.isLowerCase()) it.titlecase() else it.toString() }
    }
}
